### PR TITLE
Update Azure Go SDK version to v53.1.0

### PR DESCRIPTION
**Pull request by feiskyer**
#### What type of PR is this?

/kind cleanup
/sig cloud-provider
/area provider/azure

#### What this PR does / why we need it:

There is a security [issue](https://snyk.io/vuln/SNYK-GOLANG-GITHUBCOMSATORIGOUUID-72488) on package github.com/satori/go.uuid, which is required by Azure SDK for Go. This PR upgrades Azure Go SDK version to v53.1.0 to avoid dependency on it (the original UUID package has been replaced with "github.com/gofrs/uuid").

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Update Azure Go SDK version to v53.1.0
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by feiskyer** [comment]
@dims @liggitt  could you help to have a look at this?

**Comment by dims** [comment]
@feiskyer looks like the SDK ditched `satori/go.uuid` and went with `gofrs/go.uuid` can you please document that in the PR body?

File: staging/src/k8s.io/legacy-cloud-providers/azure/clients/loadbalancerclient/azure_loadbalancerclient_test.go [hunk 1]
```diff
@@ -172,8 +172,10 @@ func TestListWithNextPage(t *testing.T) {
 	resourceID := "/subscriptions/subscriptionID/resourceGroups/rg/providers/Microsoft.Network/loadBalancers"
 	armClient := mockarmclient.NewMockInterface(ctrl)
 	lbList := []network.LoadBalancer{getTestLoadBalancer("lb1"), getTestLoadBalancer("lb2"), getTestLoadBalancer("lb3")}
-	partialResponse, err := json.Marshal(network.LoadBalancerListResult{Value: &lbList, NextLink: to.StringPtr("nextLink")})
+	// LoadBalancerListResult.MarshalJson() doesn't include "nextLink" in its result, hence partialResponse is composed manually below.
+	mcListBytes, err := json.Marshal(lbList)
 	assert.NoError(t, err)
+	partialResponse := []byte(fmt.Sprintf(`{"value": %s, "nextLink": "nextLink"}`, string(mcListBytes)))
```
Comment (liggitt) [inline staging/src/k8s.io/legacy-cloud-providers/azure/clients/loadbalancerclient/azure_loadbalancerclient_test.go:178]: don't compose JSON content using Sprintf. xref https://github.com/kubernetes/kubernetes/issues/81134

comment applies to all three places this is being done

Comment (feiskyer) [reply staging/src/k8s.io/legacy-cloud-providers/azure/clients/loadbalancerclient/azure_loadbalancerclient_test.go:178]: thanks for the suggestions, would update per the suggestions.

**Comment by feiskyer** [comment]
/retest 

@dims @liggitt updated per comments, PTAL.

**Comment by sfowl** [comment]
any chance of cherry-picks for this? Looks like a CVE is going to be assigned to satori/go.uuid issue:

https://github.com/satori/go.uuid/issues/73

**Comment by p-rog** [comment]
FYI: the CVE-2021-3538 has been assigned to  satori/go.uuid issue.

**Pull request merged**